Doc/c-api/unicode.rst

```diff
@@ -614,6 +616,11 @@ APIs:
 
    Return the length of the Unicode object, in code points.
 
+   Check that *unicode* is a Unicode object, unlike :c:func:`PyUnicode_GET_LENGTH`,
+   which performs no error checking.
+
```

Comment:
I prefer to not explain implementation details in the C API docs. L622 should be sufficient. For the `PyUnicode_GET_LENGTH` docs, we could instead mention that it does not perform error checking.

```suggestion
```